packages/server/src/service/gm.ts

```diff
@@ -22,17 +22,11 @@ const findByType = (
 // Helping function for rawGeocode
 const geocodeQuery = (addr: AddressInputParts | string) => {
   if (typeof addr === 'object') {
-    // Using component filtering to ensure more precise results
+    // Using address + component only to filter for results in the US
     // https://developers.google.com/maps/documentation/geocoding/overview#component-filtering
-    const queryParts = [
-      'country:us',
-      `street:${addr.street}`,
-      `state:${addr.state}`,
-      `city:${addr.city}`,
-      `postcode:${addr.postcode}`,
-    ]
-    if (addr.unit) queryParts.push(`unit:${addr.unit}`)
-    return encodeURIComponent(queryParts.join(('|')))
+    const address = formatAddressInputParts(addr)
+    const components = `&components=country:us`
```

Comment:
It seems like this is the potential cause for the issues we are facing, I was using the component filtering syntax with the address param. Although it strangely worked for most queries, including the ones from the automated tests, it might have contributed to these latest unexpected behavior.

I've just finished running some local tests after rewriting the old query to use the proper components param, and I could barely get any result of it, apparently relying only on component filtering was a bad idea in the first place and its best feature for us is solely to restrain results in the USA.